core/src/nodes/collection.rs

```diff
@@ -439,6 +439,17 @@ impl<TInEvent, TOutEvent, THandler> CollectionStream<TInEvent, TOutEvent, THandl
     }
 }
 
+/// Errors for interrupt(). 
+#[derive(Debug)]
+pub enum InterruptErr {
+    /// The task entry is vacant; it needs to be added first via add_reach_attempt
+    /// (with the TaskState set to Pending) before we try to connect.
+    VacantEntry,
+    /// The task has already connected to the node; interrupting a reach attempt
+    /// is thus redundant as it has already completed.
+    Connected,
```

Comment:
Here we have a similar situation to the above: the real error is that the reach attempt ID they're trying to use is no longer valid (because it's connected). Not sure what a better name is, maybe `AlreadyReached`?